hw1-pool: init commit, implemented thread pool and tests to it

AntonYermilov opened the pull request.

AntonYermilov committed:
hw1-pool: init commit, implemented thread pool and tests to it

AntonYermilov committed:
hw1-pool: updated tests, added simple output if tests are successful

AntonYermilov committed:
hw1-pool: fixed failed test

AntonYermilov committed:
Merge branch 'master' into hw1-pool
Updating build script in hw1-pool branch

AntonYermilov committed:
hw1-pool: rewritten failed test

jzuken commented on term2/hw1-pool/src/main/java/me/eranik/threads/LightFuture.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Exception that is thrown if some error occurred when executing task.
+     */
+    class LightExecutionException extends RuntimeException {
```
ну зачем так жестоко. без особой надобности лучше не создавать непроверяемых исключений, они вносят много неявного в код

AntonYermilov replied:
Там была просто проблема в том, что в таске (которая supplier) нельзя кидать проверяемые исключения. Решил, что так ок будет. Но правда.
Fixed

jzuken commented on term2/hw1-pool/src/main/java/me/eranik/threads/ThreadPool.java:
```diff
@@ -0,0 +1,125 @@
+     */
+    public ThreadPool(int size) {
+        this.size = size;
+        this.threads= new Thread[size];
```
пробел

AntonYermilov replied:
Fixed

AntonYermilov committed:
hw1-pool:
- Fixed misprint
- LightExecutionException is checked now
- Added test for throwing LightExecutionException

AntonYermilov committed:
hw1-pool:
- Removed simple prints from tests
- Added javadocs to ThreadPool
